common/src/main/java/org/apache/rocketmq/common/BrokerConfig.java

```diff
@@ -47,6 +47,7 @@
     private boolean autoCreateSubscriptionGroup = true;
     private String messageStorePlugIn = "";
 
+    //thread counts for sending message, if large number threads are needed, we suggest setting useReentrantLockWhenPutMessage=true(in MessageStoreConfig)
```

Comment:
IMO, this is the tradeoff between two kinds of synchronization implementations: busy-waiting and mutex-lock; The former approach indeed wastes more CPU cycles which may be utilized to finish other tasks by scheduler, but it has its benefits too--doing so actually prioritize current data flow; Even if the latter is energy-saving, it has its cons, say, more latencies are introduced.  

Another factor is the operation system being used. OS is getting so smart that things may have changed considerably.

If someone may provide raw data, comparing effects of these two choices under various conditions(# of cpu cores, # of send-message-threads, flush-disk-type, etc), it would be very informative and valuable.